pkg/cluster/graph/manager_test.go

```diff
@@ -0,0 +1,79 @@
+package graph
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	azstorage "github.com/Azure/azure-sdk-for-go/storage"
+	"github.com/golang/mock/gomock"
+	"github.com/sirupsen/logrus"
+
+	mock_env "github.com/Azure/ARO-RP/pkg/util/mocks/env"
+	mock_keyvault "github.com/Azure/ARO-RP/pkg/util/mocks/keyvault"
+	mock_storage "github.com/Azure/ARO-RP/pkg/util/mocks/storage"
+	utilerror "github.com/Azure/ARO-RP/test/util/error"
+)
+
+func TestLoadPersisted(t *testing.T) {
+	ctx := context.Background()
+
+	for _, tt := range []struct {
+		name    string
+		mocks   func(*mock_storage.MockManager, *mock_env.MockInterface, *mock_keyvault.MockManager)
+		wantErr string
+	}{
+		{
+			name: "get a general error as azstorage not mocked",
+			mocks: func(storage *mock_storage.MockManager, env *mock_env.MockInterface, kv *mock_keyvault.MockManager) {
+				storage.EXPECT().BlobService(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(&azstorage.BlobStorageClient{}, nil)
+			},
+			wantErr: " authentication is not supported yet",
+		},
+		{
+			name: "loadPersisted returns an error other than the chacha20poly1305 one",
+			mocks: func(storage *mock_storage.MockManager, env *mock_env.MockInterface, kv *mock_keyvault.MockManager) {
+				storage.EXPECT().BlobService(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(&azstorage.BlobStorageClient{}, errors.New("general error"))
+			},
+			wantErr: "general error",
+		},
+		{
+			name: "loadPersisted returns a chacha20poly1305 error",
```

Comment:
Should we add one more test case for no error / `wantErr: nil`?

Reply:
the problem is that some structs like BlobStorageClient and azstorage are under vendor folder that do not have a mock interface, hence can not be properly controlled to craft a nil error case :(, guess this is why there is no test cases for graph and some cluster files before.
these few test cases are mostly covering the changes in the pr, the content of loadPersisted() are not touched this time, so it should work as before.